File: snap-crescent/src/main/java/com/codeinsight/snap_crescent/common/beans/BaseSearchCriteria.java
package com.codeinsight.snap_crescent.common.beans;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

import org.springframework.data.domain.Sort.Direction;

import com.codeinsight.snap_crescent.common.utils.Constant.ResultType;

import lombok.Data;

@Data
public class BaseSearchCriteria{

	private List<String> selectedIds = new ArrayList<>();
	
	private String searchKeyword;
	private Boolean active;
	private Date fromDate;
	private Date toDate;
	
	private String sortBy;
	private Direction sortOrder;
	
	private ResultType resultType;
	
	private Integer pageNumber = 0;
	private Integer resultPerPage = 100000;

}

File: snap_crescent_mobile-flutter/lib/models/base_search_criteria.dart

import 'package:snap_crescent/utils/constants.dart';

class BaseSearchCriteria {
  String? searchKeyword;
  bool? active;
  DateTime? fromDate;
  DateTime? toDate;

  String? sortBy;
  Direction? sortOrder;

  ResultType? resultType;

  int? pageNumber = 0;
  int? resultPerPage = 1000;

  BaseSearchCriteria(
      {this.searchKeyword,
      this.active,
      this.fromDate,
      this.toDate,
      this.sortBy,
      this.sortOrder,
      this.resultType,
      this.pageNumber,
      this.resultPerPage});

  Map<String, dynamic> toMap() {
    Map<String, dynamic> map = new Map();

    if (searchKeyword != null) {
      map['searchKeyword'] = searchKeyword;
    }

    if (active != null) {
      map['active'] = active;
    }

    if (fromDate != null) {
      map['fromDate'] = fromDate!.millisecondsSinceEpoch;
    }

    if (toDate != null) {
      map['toDate'] = toDate!.millisecondsSinceEpoch;
    }

    if (sortBy != null) {
      map['sortBy'] = sortBy;
    }

    if (sortOrder != null) {
      map['sortOrder'] = sortOrder.toString().split('.').last;
    }

    if (resultType != null) {
      map['resultType'] = resultType.toString().split('.').last;
    }

    if (pageNumber != null) {
      map['pageNumber'] = pageNumber;
    }

    if (resultPerPage != null) {
      map['resultPerPage'] = resultPerPage;
    }

    return map;
  }
}
